tensorflow/core/common_runtime/bfc_allocator.cc

```diff
@@ -260,6 +261,76 @@ size_t BFCAllocator::RoundedBytes(size_t bytes) {
   return rounded_bytes;
 }
 
+bool BFCAllocator::DeallocateFreeRegions(size_t rounded_bytes) {
+  // Searching for free regions.
+  absl::flat_hash_set<void*> free_region_ptrs;
+  size_t total_free_bytes = 0;
+  for (const auto& region : region_manager_.regions()) {
+    ChunkHandle h = region_manager_.get_handle(region.ptr());
+    bool any_use = false;
+    while (h != kInvalidChunkHandle) {
+      const Chunk* c = ChunkFromHandle(h);
```

Comment:
Calling ChunkFromHandle() requires holding mutex lock_

Reply:
Also true throughout this function.  Actually this may be best solved by taking the lock at the beginning of this function.

Reply:
Agreed.